src/rgw/rgw_rest_s3.cc

```diff
@@ -6305,34 +6323,78 @@ int RGWSelectObj_ObjStore_S3::run_s3select(const char* query, const char* input,
       csv.use_header_info=true;
     }
 
-    m_s3_csv_object = std::unique_ptr<s3selectEngine::csv_object>(new s3selectEngine::csv_object(s3select_syntax.get(), csv));
+    if (m_parquet_type){
+      m_s3_parquet_object = std::unique_ptr<s3selectEngine::parquet_object>(new s3selectEngine::parquet_object(s->object->get_name(), s3select_syntax.get(), m_rgw_api));
+    } 
+    else {
+      m_s3_csv_object = std::unique_ptr<s3selectEngine::csv_object>(new s3selectEngine::csv_object(s3select_syntax.get(), csv));
+    }
   }
 
   header_size = create_header_records(m_buff_header.get());
   m_result.append(m_buff_header.get(), header_size);
   m_result.append(PAYLOAD_LINE);
 
-  if (s3select_syntax->get_error_description().empty() == false) {
+  std::function<int(std::string&)> fp_result_header_format = [&](std::string& result) {
+    
+    result = "012345678901";
+    header_size = create_header_records(m_buff_header.get());
+    result.append(m_buff_header.get(), header_size);
+    result.append(PAYLOAD_LINE);
+    return 0;
+  };
+
+  std::function<int(std::string&)> fp_s3select_result_format = [&](std::string &result) {
+
+    result.append(END_PAYLOAD_LINE);
+    int buff_len = create_message(result, result.size() - 12, header_size);
+    s->formatter->write_bin_data(result.data(), buff_len);
+    rgw_flush_formatter_and_reset(s, s->formatter);
+
```

Comment:
I'm not sure you need this whitespace, either

Reply:
:+1: